app/models/wpcc/your_contribution.rb

```diff
@@ -4,5 +4,20 @@ class YourContribution
     attr_accessor :eligible_salary,
                   :employee_percent,
                   :employer_percent
+
+    CONTRIBUTION_PREFERENCES = {
+      'minimum' => MinimumContributionCalculator,
+      'full' => FullContributionCalculator
+    }.freeze
+
+    def self.generate(salary, contribution_preference)
+      contribution_klass = CONTRIBUTION_PREFERENCES[contribution_preference]
+      calculator = contribution_klass.new(salary, contribution_preference)
+      new(
+        eligible_salary:  calculator.eligible_salary,
+        employee_percent: calculator.employee_percent,
+        employer_percent: calculator.employer_percent
+      )
+    end
```

Comment:
That makes 2 things - selecting the right class and instantiating a new instance. Any method that is greater than 5 lines long is a candidate for refactoring and usually points to more than one responsibility.